pipert/core/main.py

```diff
@@ -0,0 +1,87 @@
+import os
+from flask import Flask, jsonify, request, Response
+from pipert.core.pipeline_manager import PipelineManager
+import inspect
+import zerorpc
+
+
+class CliConnection(object):
+
+    def __init__(self, pipeline_manager):
+        self.pipeline_manager = pipeline_manager
+
+    def execute_method(self, method_name, parameters_values):
+        return getattr(self.pipeline_manager, method_name)(**parameters_values)
+
+    def get_methods(self):
+        methods = inspect.getmembers(self.pipeline_manager, predicate=inspect.ismethod)
+        methods = [method[0] for method in methods if (not method[0].startswith('_'))]
+        return methods
+
+    # need to be tested
+    def get_method_parameters(self, method_name):
+        return list(inspect.signature(getattr(self.pipeline_manager, method_name)).parameters.keys())
+
+    def check_connection(self):
+        return True
+
+
+# # use_user_interface = os.environ.get("UI").lower() == 'true'
+use_user_interface = input("Do you want to use the UI ? (y/n): ").lower() == 'y'
```

Comment:
What is your thoughts about putting it as an environment param in the docker-compose?

Reply:
In my opinion when we build the pipeline using the docker-compose we should pass a variable conatining whether we want the ui or the cli, it seems to me that it is possible but i am not familier with it.
somthing like that: docker-compose up -d -ui True/False
You have more experience with it and maybe you can come with a better idea.

Also, same thing for the endpoint, if we decide to use the cli we should pass the endpoint of the zerorpc server as a parameter

Reply:
There are environment variables that we pass to the pipe from the docker-compose.
Add this as another one and check the value with os.environ.get()